r/src/matrix_utils.cpp

```diff
@@ -624,6 +623,32 @@ List apply_matrix_double_cpp(SEXP mat_sexp, Function f, bool row_major) {
     return ret;
 }
 
+// Compute the max of each row
+// [[Rcpp::export]]
+NumericVector matrix_compute_max_per_row(SEXP matrix) {
```

Comment:
I'd suggest naming these `matrix_max_per_row_cpp` and `matrix_max_per_col_cpp`. I've been puting the `_cpp` ending on all Rcpp functions to make it clear where in the R code is calling into C++. And I think the "compute" part of the name is not adding a lot of information to the name (though that's just a matter of taste)